addons/binding/org.openhab.binding.fronius/src/main/java/org/openhab/binding/fronius/internal/model/ActiveDeviceInfo.java

```diff
@@ -0,0 +1,171 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.fronius.internal.model;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import org.eclipse.smarthome.core.library.types.DateTimeType;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
+/**
+ * Access object for device information.
+ *
+ * @author Gerrit Beine
+ */
+public class ActiveDeviceInfo {
+
+    private final Logger logger = LoggerFactory.getLogger(ActiveDeviceInfo.class);
+    private final Set<Integer> inverters = new HashSet<>();
+    private final Set<Integer> storages = new HashSet<>();
+    private final Set<Integer> meters = new HashSet<>();
+    private final Set<Integer> ohmpilots = new HashSet<>();
+    private final Set<Integer> sensorCards = new HashSet<>();
+    private final Set<Integer> stringControls = new HashSet<>();
+
+    private DecimalType code = DecimalType.ZERO;
+    private DateTimeType timestamp = new DateTimeType();
+    private boolean empty = true;
+
+    public static ActiveDeviceInfo createActiveDeviceInfo(final JsonObject json) {
+        final ActiveDeviceInfo adi = new ActiveDeviceInfo();
+        adi.deconstruct(json);
+        return adi;
+    }
+
+    private ActiveDeviceInfo() {
+        super();
+    }
+
+    public boolean isEmpty() {
+        return this.empty;
+    }
+
+    public int inverterCount() {
+        return inverters.size();
+    }
+
+    public int meterCount() {
+        return meters.size();
+    }
+
+    public int storageCount() {
+        return storages.size();
+    }
+
+    public int ohmpilotCount() {
+        return ohmpilots.size();
+    }
+
+    public int sensorCardCount() {
+        return sensorCards.size();
+    }
+
+    public int stringControlCount() {
+        return stringControls.size();
+    }
+
+    public DateTimeType getTimestamp() {
+        return timestamp;
+    }
+
+    private synchronized void deconstruct(final JsonObject json) {
+        try {
```

Comment:
Any reason to deconstruct the json yourself instead of using gson to deserialize it into data classes?